src/Symfony/Component/Lock/Store/DatabaseTableTrait.php

```diff
@@ -48,8 +48,12 @@ private function init(array $options, float $gcProbability, int $initialTtl): vo
     /**
      * Returns a hashed version of the key.
      */
-    private function getHashedKey(Key $key): string
+    private function getKeyName(Key $key): string
     {
+        $keyAsString = (string) $key;
+        if (strlen($keyAsString) <= 64) {
+            return $keyAsString;
+        }
         return hash('sha256', (string) $key);
```

Comment:
It'd keep the original name + leverage PHP to do the cast:
```suggestion
    private function getHashedKey(string $key): string
    {
        return strlen($keyAsString) <= 64 ? $key : hash('sha256', $key);
```

Reply:
Thanks!